src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3641,6 +3647,18 @@ float ViewProviderSketch::getScaleFactor()
     }
 }
 
+void ViewProviderSketch::InitItemsSizes()
+{
+    int defaultFontSize = QApplication::fontMetrics().height();
+    int ldpi = QApplication::desktop()->logicalDpiX();
+    float k = 96./ldpi;
+    float view3D_factor = 1.25;
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/View");
+    coinFontSize = hGrp->GetInt("EditSketcherFontSize", defaultFontSize * k * k *view3D_factor );

```

Comment:
also here, please add a comment why e.g. 'k' appears twice.

Reply:
the ratios are as follows (exclude factor):
iconsize = defaulfontsize * k;
coinFontSize = iconsize * k;
since the settings store the font size, we first set it, and then the icon size
coinFontSize = defaulfontsize * k *k  (or from settings);
iconsize = coinFontSize / k;

Reply:
Thanks. Can you please add this info via comments and give the variable a more meaningful name than 'k'?
For example, why do we use here 96 as DPi setting, not 90 or 92? I guess this is hardcoded somewhere and it would be good to know where and why.
Also for the 'k*k', please at a comment line above that coinFontSize = defaultfontsize * k *k

I don't want to annoy you, I only think that code should be self-explaining for other programmers. So just add as many comments as you like to achieve this. We don't have a comment limit ;-)

Reply:
I don't see you're taking into account the device pixel ratio anywhere.

Does Coin view use global scaling? It would mean that icons would be pixelated when DPR is not 100%.
Not a blocker for this PR but something to keep in mind for the future.

Also, please rename `k` to `devicePixelRatio`.

Reply:
Here's what I'd do:
```
qreal ViewProviderSketch::ptToPx(qreal pt) {
    int ldpi = QApplication::desktop()->logicalDpiX();
    return pt / 72 * ldpi;
}

qreal ViewProviderSketch::getCoinFontSizePx() {
  qreal sketcherFontSizeRem = 1.25; // a little bit bigger than the default size
  qreal defaultFontSizePt = QApplication::font().pointSizeF();
  qreal defaultSketcherFontSizePt = defaultFontSizePt * sketcherFontSizeRem;

  qreal sketcherFontSizePt = hGrp->GetInt("EditSketcherFontSizePt", defaultSketcherFontSizePt);
  qreal coinFontSizePx = ptToPx(sketcherFontSizePt) * devicePixelRatio();
  return coinFontSizePx;
}

qreal ViewProviderSketch::getConstrainIconSizePx() {
  return getCoinFontSizePx() * 0.8; // a little smaller than the sketcher font size
}
```

Reply:
> Thanks. Can you please add this info via comments and give the variable a more meaningful name than 'k'?
corrected.
I hope this will be enough to start understanding.
To dive into the problem (when you need to edit), you still need to know about Qt's scaling methods, units of measure for fonts and images, etc.